server/core/src/storage.ts

```diff
@@ -138,16 +139,50 @@ class TServerStorage implements ServerStorage {
     return adapter
   }
 
-  private async routeTx (ctx: MeasureContext, tx: Tx): Promise<TxResult> {
-    if (this.hierarchy.isDerived(tx._class, core.class.TxCUD)) {
-      const txCUD = tx as TxCUD<Doc>
+  private async routeTx (ctx: MeasureContext, ...txes: Tx[]): Promise<TxResult> {
+    let part: TxCUD<Doc>[] = []
+    let lastDomain: Domain | undefined
+    const result: TxResult[] = []
+    const processPart = async (): Promise<void> => {
+      if (part.length > 0) {
+        const adapter = this.getAdapter(lastDomain as Domain)
+        const r = await adapter.tx(...part)
+        if (Array.isArray(r)) {
+          result.push(...r)
+        } else {
+          result.push(r)
+        }
+        part = []
+      }
+    }
+    for (const tx of txes) {
+      const txCUD = TxProcessor.extractTx(tx) as TxCUD<Doc>
+      if (!this.hierarchy.isDerived(txCUD._class, core.class.TxCUD)) {
+        // Skip unsupported tx
+        console.error('Unsupported transacton', tx)
+        continue
+      }
       const domain = this.hierarchy.getDomain(txCUD.objectClass)
-      const adapter = this.getAdapter(domain)
-      const res = await adapter.tx(txCUD)
-      return res
-    } else {
+      if (part.length > 0) {
```

Comment:
```suggestion
     if (lastDomain !== domain && part.length > 0) {
await processPart()
}
lastDomain = domain
part.push(txCUD)
```

Reply:
I do no think this check is really required, in place there it is needed it checked.